java/maven/src/org/netbeans/modules/maven/execute/cmd/ShellConstructor.java

```diff
@@ -44,7 +44,9 @@ public ShellConstructor(@NonNull File mavenHome) {
     public List<String> construct() {
 
         // use mvnd if its the home of a daemon
-        String ex = Files.exists(Paths.get(mavenHome.getPath(), "bin", "mvnd")) ? "mvnd" : "mvn"; //NOI18N
+        String mavenDaemonSuffixDetection = Utilities.isWindows() ? ".cmd" : "";
+        String ex = Files.exists(Paths.get(mavenHome.getPath(), "bin", "mvnd" + mavenDaemonSuffixDetection)) ? "mvnd" : "mvn"; //NOI18N
```

Comment:
i think we should probe here for `mvnd.exe` instead of `.cmd` because `mvnd` is a binary. It is not super important but it would make it symmetrical.

Thanks for fixing this for windows btw. Feel free to force push.

Reply:
Change done :)